src/pages/index.js

```diff
@@ -100,36 +148,123 @@ function fillProfileForm() {
     descInput.value = info.description;
 }
 
-function updateInfo(event, data) {
-    userInfo.setUserInfo({
+async function updateInfo(event, data) {
+    const newInfo = {
         name: data["name-input"],
         description: data["description-input"],
+    };
+    userInfo.setUserInfo(newInfo);
+    api.editProfile({
+        name: newInfo.name,
+        about: newInfo.description
+    })
+    .catch((err) => {
+        console.log("Error updating user info: " + err);
     });
 }
 
+function openAvatarModal() {
+    avatarPopup.open();
+    formValidators["avatar-form"].toggleButtonState();
+}
+
+async function updateAvatar(event, data) {
+    const link = data["url-input"];
+
+    api.updateAvatar(link)
+        .then((res) => {
+            userInfo.setUserAvatar(res.avatar);
+        })
+        .then(() => {
+            event.target.reset();
+        })
+        .catch((err) => {
+            console.log("Error updating avatar: " + err);
+        });
+    
+}
+
 // #endregion 
 
-// #region New place modal & card creation methods
+// #region New place modal & card methods
 
 function openPlaceModal() {
     formValidators["place-form"].toggleButtonState();
     placePopup.open();
 }
 
-function addCard(event, data) {
+function openConfirmModal(event, data) {
+    deletePopup.open();
+    deletePopup.setConfirmationData({
+        cardElement: event.target.closest(".card"),
+        data: data,
+    });
+}
+
+async function addCard(event, data) {
     const cardData = {
         name: data["title-input"],
         link: data["url-input"],
     };
-    cardsSection.addItem(getCardElement(cardData));
-    event.target.reset();
+
+    api.addCard(cardData)
+        .then((res) => {
+            cardData._id = res._id;
+            cardsSection.addItem(getCardElement(cardData));
+        })
+        .then(() => {
+            event.target.reset();
+        })
+        .catch((err) => {
+            console.log("Error adding a card: " + err);
+        });
+}
+
+async function deleteCard({cardElement, data}) {
+    api.deleteCard(data._id)
+        .then(() => {
+            cardElement.remove();
+        })
+        .catch((err) => {
+            console.log("Error deleting a card: " + err);
+        });
+    deletePopup.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well